flytekit/core/array_node.py

```diff
@@ -49,6 +50,7 @@ def __init__(
         self._concurrency = concurrency
         self._execution_mode = execution_mode
         self.id = target.name
+        self._bindings: List[_literal_models.Binding] = []
```

Comment:
Make this optional?  and set default to None.

Reply:
@wild-endeavor would it make sense to keep this as is since this isn't a passed in value and to have:

```
    @property
    def bindings(self) -> List[_literal_models.Binding]:
        # Required in get_serializable_node
        return self._bindings
```

return an empty list if the bindings aren't set

Reply:
see slack